compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -690,11 +702,28 @@ fn report_missing_placeholders(
     if !found_foreign && unused.len() == 1 {
         diag.span_label(fmt_span, "formatting specifier missing");
     }
-    if dup_exist {
-        diag.note("the formatting string captures that binding directly, it doesn't need to be included in the argument list");
-        let this_these = if duplicate_explicit_arg.len() == 1 { "this" } else { "these" };
-        diag.span_help(duplicate_explicit_arg, format!("Consider removing {}", this_these));
+    if !redundant_expl_args.is_empty() {
+        let mut fmt_spans = Vec::new();
+        let mut arg_spans = Vec::new();
+        let mut arg = Vec::new();
+        for rd in &redundant_expl_args {
+            fmt_spans.push(Span::from_inner(fmt_span, rd.fmt_span));
+            arg_spans.push(rd.arg_span);
+            arg.push(rd.fmt_str);
+        }
+        let mut m_span: MultiSpan = fmt_spans.clone().into();
+        for i in 0..fmt_spans.len() {
+            m_span.push_span_label(
+                fmt_spans[i],
+                format!("this formatting argument captures `{}` directly", arg[i]),
+            );
+        }
+        for arg_span in arg_spans {
+            m_span.push_span_label(arg_span, "this can be removed".to_string());
+        }
+        diag.span_help(m_span, "the formatting string captures that binding directly, it doesn't need to be included in the argument list");
```

Comment:
We should customize the output depending on the number of bindings:
```suggestion
        let mut bindings: Vec<_> = arg.iter().collect();
        bindings.sort();
        bindings.dedup();
        diag.span_help(
            m_span,
            &format!(
                "the formatting string captures {} directly, {} doesn't need to \
                 be included in the argument list",
                match &bindings[..] {
                    [.., last] if bindings.len() < 8 => format!("bindings `{}` and `{last}`"),
                    [binding] => format!("binding {binding}"),
                    _ => "these bindings".to_string(),
                },
                if bindings.len() == 1 {
                    "it"
                } else {
                    "they"
                },
            );
```

Reply:
This request is still valid :)

You can choose to simplify it, but at least I'd like to specialize the output for a single binding as opposed to multiple.